File: flutter-app/lib/models/report.dart
class Report {
  final int id;
  final String? title;
  final String? description;
  final String? status;
  final String? createdAt;

  const Report({
    required this.id,
    this.title,
    this.description,
    this.status,
    this.createdAt,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      id: json['id'] as int,
      title: json['title'] as String?,
      description: json['description'] as String?,
      status: json['status'] as String?,
      createdAt: json['createdAt'] as String?,
    );
  }
}

File: ios-app/UniSloveApp/UniSloveApp/Models/Report.swift
import Foundation

struct Report: Decodable, Identifiable {
    let id: String
    let title: String?
    let description: String?
    let category: String?
    let status: String?
    let userRating: Int?
    let isApproved: Bool?
    let createdAt: String?
    let user: UserProfile?
    let likes: [APILikeItem]?
    let comments: [CommentItem]?
}
